### PR TITLE
Update Action framework usage docs to reflect unsupported design option

### PR DESCRIPTION
**Pull request by vinaykul**
**What type of PR is this?**
> Uncomment only one ` /kind <>` line, hit enter to put that in a new line, and remove leading whitespaces from that line:
>
> /kind api-change
> /kind bug
> /kind cleanup
> /kind design
/kind documentation
> /kind failing-test
> /kind feature
> /kind flake

**What this PR does / why we need it**: Action framework recommends using subresource pattern to create actions for simplicity. Alternate mechanism mentioned in the design doc is not supported, and this change clarifies that fact.

**Which issue(s) this PR fixes**: #319 

Fixes #

**Special notes for your reviewer**:

**Does this PR introduce a user-facing change?**:

```release-note

```

### CODE REVIEW

File: docs/design-proposals/apiserver/Action-API-design.md [hunk 1]
```diff
@@ -94,10 +94,42 @@ specify a desired action on Pod object. This is the recommended way for user
...
+{
+  "apiVersion": "v1",
+  "kind": "CustomAction",
+  "actionName": "Reboot",
+  "rebootParams": {
+    "delayInSeconds": 10
+  }
+}
+
+root@fw0000358:~/ActAlk# curl -X POST http://127.0.0.1:8001/api/v1/namespaces/default/pods/1pod/action -H "Content-Type: application/json" -d @../YML/reboot.json
```
Comment (yb01) [inline docs/design-proposals/apiserver/Action-API-design.md:110]: this curl call is used at the time kubectl is not supported. now kubectl is updated so we should change to use kubectl create -f action.yaml

File: docs/design-proposals/apiserver/Action-API-design.md [hunk 2]
```diff
@@ -94,10 +94,42 @@ specify a desired action on Pod object. This is the recommended way for user
...
+{
+  "apiVersion": "v1",
+  "kind": "CustomAction",
+  "actionName": "Snapshot",
+  "snapshotParams": {
+    "snapshotLocation": "/var/tmp/"
+  }
+}
+
+root@fw0000358:~/ActAlk# curl -X POST http://127.0.0.1:8001/api/v1/namespaces/default/pods/1pod/action -H "Content-Type: application/json" -d @../YML/snapshot.json
```
Comment (yb01) [inline docs/design-proposals/apiserver/Action-API-design.md:125]: same here.

File: docs/design-proposals/apiserver/Action-API-design.md (hunk 1 above)
Comment (vinaykul) [reply docs/design-proposals/apiserver/Action-API-design.md:110]: We don't support create action now. The above curl is supported - which would translate to Action subresource when done programmatically.

File: docs/design-proposals/apiserver/Action-API-design.md (hunk 2 above)
Comment (vinaykul) [reply docs/design-proposals/apiserver/Action-API-design.md:125]: this is illustration of action subresource.

**Comment by vinaykul** [comment]
reopen

**Pull request closed without merging**